src/utils/scales-utils.js

```diff
@@ -363,7 +363,13 @@ function _computeScaleDistance(values, domain, bestDistIndex, scaleFn) {
  */
 export function _getScaleDistanceAndAdjustedDomain(data, scaleObject) {
   const {attr, domain} = scaleObject;
+
   const values = getUniquePropertyValues(data, attr);
+  // Fix time scale if a data has only one value.
+  if (scaleObject.type === TIME_SCALE_TYPE && values.length === 1) {
```

Comment:
I still think this should be a helper method. It'll make the code easier to read and can avoid mutating data